### PR TITLE
code-update:writing unit test for user service

### PR DESCRIPTION
**Pull request by vanshika-dargan-18**
Task Overview:

Added uts for user module crud operations

In order to run uts manually

```yarn run test:watch```

cc: @pratikjoshivonage @Dys-Mn11 @Shubham-rajput09

### CODE REVIEW

File: src/users/users.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    userRepository = module.get(getRepositoryToken(UserEntity));
+    passwordService = module.get(PasswordService);
+  });
+
+  it('should be defined', () => {
+    expect(userService).toBeDefined();
+  });
+
+  // create user
+  describe('create', () => {
```
Comment (mgolchha) [inline src/users/users.service.spec.ts]: Add tests to handle error cases, such as when a user is not found or when the repository methods throw exceptions.

**Review by mgolchha — COMMENTED** [review]
Ensure consistency in mock return values across tests. For example, mockResolvedValue and mockReturnValue are used; consider standardizing these for clarity.

Test additional edge cases, such as creating a user with an existing email, updating a non-existent user, or deleting a user that doesn’t exist.

File: src/users/users.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        ...rest,
+        password: 'hashed_abcdef',
+      });
+      expect(userRepository.save).toHaveBeenCalledWith(user);
+      expect(result).toEqual(user);
+    });
+  });
+
+  //get all users
+  describe('findAll', () => {
```
Comment (pratikjoshivonage) [inline src/users/users.service.spec.ts]: Please cover both the scenarios. If we get data or if we get empty data.
  describe('findAll with data returned', () => {

  describe('findAll with empty data returned', () => {

File: src/users/users.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+      expect(result).toEqual(users);
+    });
+  });
+
+  // get user by id
+  describe('findById', () => {
+    it('should return a user by id', async () => {
+      jest.spyOn(userRepository, 'findOne').mockResolvedValue(user);
+      const result = await userService.findById(1);
+      expect(userRepository.findOne).toHaveBeenCalledWith({
```
Comment (pratikjoshivonage) [inline src/users/users.service.spec.ts]: Here we can cover if data is found, data is not found.

**Review by pratikjoshivonage — CHANGES_REQUESTED** [review]

File: src/users/users.service.spec.ts (hunk 3 above)
Comment (vanshika-dargan-18) [reply src/users/users.service.spec.ts]: @pratikjoshivonage pls check once now.

**Review by pratikjoshivonage — APPROVED** [review]

**Review by Shubham-rajput09 — APPROVED** [review]
